scripts/fbi/hate_crime/download_publication_data.py

```diff
@@ -0,0 +1,363 @@
+    else:
+        resp_html = None
+        resp_soup = None
+
+        print('HTTP status code: ' + str(req.status_code))
+    return resp_html, resp_soup
+
+
+def _find_year_url(html_soup, year_str: str) -> str:
+    """Find the URL for a given year from the base HTML list of URL for all years.
+    Args:
+        html_soup: BeautifulSoup object of the HTML page with year URL.
+        year_str: string version of the year for which the URL is to be found.
+    Returns:
+        URL to the page for the data of particular year.
+  """
+    year_url = None
+    base_a_list = html_soup.find(id='content-core').find_all('a')
+    for cur_a in base_a_list:
+        cur_year = cur_a.string
+        if cur_year == year_str:
+            print(cur_year)
+            year_url = cur_a.get('href')
+    return year_url
+
+
+def _find_access_tables_url() -> str:
+    """Find the URL to 'Access Tables' in the given HTML for the year.
+  
+  The method uses status variables set in the status_dict for context.
+    Returns:
+        URL to the page with list of all the tables for a particular year.
+  """
+    global _STATUS_DICT
+    # fetch or return from cache
+    if _STATUS_DICT['force_fetch'] or not os.path.isfile(
+            os.path.join(_STATUS_DICT['cache_dir'],
+                         f"{_STATUS_DICT['cur_year']}.html")):
+        year_html, year_soup = request_html_soup(_STATUS_DICT['session'],
+                                                 _STATUS_DICT['cur_year_url'])
+        with open(
+                os.path.join(_STATUS_DICT['cache_dir'],
+                             f"{_STATUS_DICT['cur_year']}.html"), 'w') as fp:
+            fp.write(year_html)
+    else:
+        with open(
+                os.path.join(_STATUS_DICT['cache_dir'],
+                             f"{_STATUS_DICT['cur_year']}.html"), 'r') as fp:
+            year_html = fp.read()
+            year_soup = BeautifulSoup(year_html, features="html.parser")
+
+    # some year pages contain and iframe which has the actual html content,
+    # 1st iframe is used if multiple are found
+    year_iframe_list = year_soup.find_all('iframe')
+    if len(year_iframe_list) > 0:
+        _STATUS_DICT['cur_year_url'] = year_iframe_list[0]['src']
+        print('iframe', _STATUS_DICT['cur_year_url'])
+        year_html, year_soup = request_html_soup(_STATUS_DICT['session'],
+                                                 _STATUS_DICT['cur_year_url'])
+        with open(
+                os.path.join(_STATUS_DICT['cache_dir'],
+                             f"{_STATUS_DICT['cur_year']}.html"), 'w') as fp:
+            fp.write(year_html)
+    # convert for use of base URL in the next step
+    _STATUS_DICT['cur_year_url'] = _STATUS_DICT[
+        'cur_year_url'][:_STATUS_DICT['cur_year_url'].rfind('/')]
+    year_subitem_list = year_soup.find_all('div', class_='subitem')
+    access_tables_found = False
+    access_url = None
+    if len(year_subitem_list) > 0:
```

Comment:
can remove 160 so save an indent level as the for loop will automatically skip if list is empty

Reply:
fixed.